src/libcore/cell.rs

```diff
@@ -187,6 +187,29 @@ use ops::{Deref, DerefMut, CoerceUnsized};
 use ptr;
 
 /// A mutable memory location.
+/// 
+/// ```
+/// use std::cell::Cell;
+///
+/// struct SomeStruct {
+///     regular_field: u8,
+///     special_field: Cell<u8>,
+/// }
+///
+/// let my_struct = SomeStruct {
+///     regular_field: 0,
+///     special_field: Cell::new(1),
+/// };
+///
+/// let new_value = 100;
+/// 
+/// // ERROR, because my_struct is immutable
+/// // immutable.regular_field = new_value;
+/// 
+/// // WORKS, special_field is mutable because it is Cell
+/// immutable.special_field.set(new_value);
+/// assert_eq!(immutable.special_field.get(), new_value);
+/// ```
```

Comment:
I don't think this follows the prevailing style in the docs. For example, I think it is at least missing an `Examples` header. I'd also like to see some prose describing what the example is trying to show.

cc @steveklabnik